src/raglite/_rag.py

```diff
@@ -134,27 +129,61 @@ async def async_rag(  # noqa: PLR0913
     """Retrieval-augmented generation."""
     # Get the contexts for RAG as contiguous segments of chunks.
     config = config or RAGLiteConfig()
-    segments = _contexts(
+    segments = context_segments(
         prompt,
         max_contexts=max_contexts,
         context_neighbors=context_neighbors,
         search=search,
         config=config,
     )
-    system_prompt = f"{system_prompt}\n\n" + "\n\n".join(
-        f'<context index="{i}">\n{segment.strip()}\n</context>'
-        for i, segment in enumerate(segments)
+    messages = _compose_messages(
+        prompt=prompt, system_prompt=system_prompt, messages=messages, segments=segments
     )
     # Stream the LLM response.
-    async_stream = await acompletion(
-        model=config.llm,
-        messages=[
-            *(messages or []),
-            {"role": "system", "content": system_prompt},
-            {"role": "user", "content": prompt},
-        ],
-        stream=True,
-    )
+    async_stream = await acompletion(model=config.llm, messages=messages, stream=True)
     async for output in async_stream:
         token: str = output["choices"][0]["delta"].get("content") or ""
         yield token
+
+
+def _compose_messages(
+    prompt: str,
+    system_prompt: str,
+    messages: list[dict[str, str]] | None,
+    segments: list[ContextSegment] | None,
+    context_placement: Literal[
+        "system_prompt", "user_prompt", "separate_system_prompt"
+    ] = "user_prompt",
+) -> list[dict[str, str]]:
```

Comment:
I understand the idea to make this configurable, but I would prefer to have RAGLite provide an opinionated default with minimal configuration instead. Here, I think the `user_prompt` strategy is probably the best solution. If you agree, we can drop the other code paths.